Objects/abstract.c

```diff
@@ -2377,11 +2377,17 @@ abstract_issubclass(PyObject *derived, PyObject *cls)
     PyObject *bases = NULL;
     Py_ssize_t i, n;
     int r = 0;
+    int derived_ref = 0;
```

Comment:
Perhaps just take a ref of `derived` before the loop, and remove this variable and its checks altogether? My intention with `derived_ref` was to avoid unnecessary memory writes to a shared/common location.

Reply:
Yes, it may simplify the code, at cost of making it slightly slower. But this is not performance critical code, it is only called when any of arguments is not a type (very unusual case).

Alternatively, you can keep a reference to `bases` until you get a new `bases`. `Py_SETREF` may help to keep the code shorter: `Py_SETREF(bases, abstract_get_bases(derived))`, but it may still look less obvious.

Left it on to you. All variants are good enough.

Reply:
`Py_SETREF` seems nice. I didn't know of this macro. I'll use it.

Reply:
It is not public. Don't use it in third-party code. It is safer to copy its definition under other name if you need it.

Reply:
Sorry, I don't think I understood that last comment of yours. What do you mean by third-party code? Using it here is okay, right?

Reply:
In your own code, in the code of your company or in the code of other projects except CPython.

It is not the part of the public C API and can be renamed, changed or removed in new Python versions.

Reply:
Mhm, I see. Thanks for the tip!